lib/rules/no-extra-boolean-cast.js

```diff
@@ -91,7 +97,14 @@ module.exports = {
                     context.report({
                         node,
                         message: "Redundant Boolean call.",
-                        fix: fixer => fixer.replaceText(node, sourceCode.getText(node.arguments[0]))
+                        fix: fixer => {
+                            const argument = node.arguments[0];
+
+                            if (astUtils.getPrecedence(argument) < astUtils.getPrecedence(node)) {
```

Comment:
It looks like this is using the precedence of the `Boolean()` call expression, but remember that the call expression won't exist anymore when `Boolean` is removed. Instead, I think we want to compare it to `node.parent`, which is a `UnaryExpression`.

To double-check, I think it would be a good idea to add cases like these to the tests:

```js
!Boolean(foo + bar);

// fixed to

!(foo + bar);

// ---

!Boolean(+foo);

// fixed to

!+foo;

// ---

!Boolean(foo());

// fixed to

!foo();

// ---

!Boolean(foo = bar);

// fixed to

!(foo = bar);
```